zipline/gens/tradesimulation.py

```diff
@@ -100,6 +101,16 @@ def transform(self, stream_in):
                 self.simulation_dt = date
                 self.on_dt_changed(date)
 
+                # removing expired futures
+                for sid in self.current_data.keys():
+                    try:
+                        if self.env.asset_finder.retrieve_asset(sid).end_date \
+                                < self.simulation_dt:
+                            self.current_data.__delitem__(sid)
+                    except (AttributeError, TypeError, ValueError,
```

Comment:
Are the keys of `current_data` just sids (meaning, integers rather than asset objects)?  I was under the impression that they should always be `Asset` objects, in which case we shouldn't need to do a `retrieve_asset` call here.  I'm nervous about the performance implications of looping over all of `data` here.  This is basically the inner loop of the simulator, so small changes here can have major performance implications.